test_integration/1_orchestrator_basic_config_suite_test.go

```diff
@@ -0,0 +1,137 @@
+package test_integration
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/suite"
+	"strings"
+	"testing"
+)
+
+type OrchestratorBasicConfigSuite struct {
+	BaseDockerComposeTestSuite
+}
+
+func NewOrchestratorBasicConfigSuite() *OrchestratorBasicConfigSuite {
+	s := &OrchestratorBasicConfigSuite{}
+	s.GlobalRunIdentifier = globalTestExecutionId
+	s.SuiteRunIdentifier = strings.ToLower(strings.Split(uuid.New().String(), "-")[0])
+	return s
+}
+
+func (s *OrchestratorBasicConfigSuite) SetupSuite() {
+	rawDockerComposeFilePath := "./common_assets/docker_compose_files/orchestrator-node-with-custom-start-command.yml"
+	s.Context, s.Cancel = context.WithCancel(context.Background())
+
+	orchestratorConfigFile := s.commonAssets("nodes_configs/1_basic_orchestrator_config.yaml")
+	orchestratorStartCommand := fmt.Sprintf("bacalhau serve --config=%s", orchestratorConfigFile)
+	extraRenderingData := map[string]interface{}{
+		"OrchestratorStartCommand": orchestratorStartCommand,
+	}
+	s.BaseDockerComposeTestSuite.SetupSuite(rawDockerComposeFilePath, extraRenderingData)
+
+	_, err := s.executeCommandInDefaultJumpbox(
+		[]string{
+			"mkdir",
+			"-p",
+			fmt.Sprintf("/app/%s", s.SuiteRunIdentifier),
+		})
+	s.Require().NoErrorf(err, "Error creating a suite tmp folder in the jumpbox node: %q", err)
+}
+
+func (s *OrchestratorBasicConfigSuite) TearDownSuite() {
+	s.T().Log("Tearing down [Test Suite] in OrchestratorBasicConfigSuite...")
+	s.BaseDockerComposeTestSuite.TearDownSuite()
+}
+
+func (s *OrchestratorBasicConfigSuite) TestNodesCanBeListed() {
+	nodeListOutput, err := s.executeCommandInDefaultJumpbox([]string{"bacalhau", "node", "list", "--output=json"})
+	s.Require().NoErrorf(err, "Error listing nodes: %q", err)
+
+	marshalledOutput, jsonType, err := s.unmarshalJSONString(nodeListOutput)
+	s.Require().NoErrorf(err, "Error unmarshalling json string: %q. JSON output received: %q", err, nodeListOutput)
+	s.Require().Equalf("array", jsonType, "incorrect json output type")
+
+	marshalledOutputArray := marshalledOutput.([]interface{})
+	s.Require().Equalf(1, len(marshalledOutputArray), "Node count should be 1")
```

Comment:
can we have a helper method to unmarshall the output into a go type? In this case the output I believe is `[]models.NodeState`

Reply:
I am looking for a way to nicely access the returned response, in a "python style" non verbose manner, and I agree with you it is currently done in a very verbose way. 

On the other hand, I am intentionally shying away from importing any code from the Bacalhau main module, including the models. My thinking is that these integration tests should only interface with the system using the CLI or SDk, or API. I feel if we open that door, a lot of the main module code will leak into the tests, making them very toughly coupled, and not focused on testing the outer contract.

Reply:
> I am intentionally shying away from importing any code from the Bacalhau main module

I agree with this intention, but unfortunately our current code is not structured to enable this just yet. I mean our API and SDK depend directly on the `models` package inside the main pkg. Besides, our go SDK is actually part of the main module where the server and all backend code lives. Meaning you can't use the SDK without importing the whole thing, which is totally wrong. 

We need to fix this, but we don't have to fix it today. Most likely we will need to have a shared module for model definitions that are shared by the SDK, API, CLI and backend.  So my recommendation for the integration tests is to use all types the SDK and API models are accessing regardless where they live and those will be migrated eventually outside of the core module